recipes/userver/all/conanfile.py

```diff
@@ -0,0 +1,655 @@
+    license = 'Apache-2.0'
+    package_type = "library"
+
+    settings = 'os', 'arch', 'compiler', 'build_type'
+    options = {
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'lto': [True, False],
+        'with_jemalloc': [True, False],
+        'with_mongodb': [True, False],
+        'with_postgresql': [True, False],
+        'with_postgresql_extra': [True, False],
+        'with_redis': [True, False],
+        'with_grpc': [True, False],
+        'with_clickhouse': [True, False],
+        'with_rabbitmq': [True, False],
+        'with_utest': [True, False],
+        'namespace': ['ANY'],
+        'namespace_begin': ['ANY'],
+        'namespace_end': ['ANY'],
+    }
+
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'lto': True,
+        'with_jemalloc': False,
+        'with_mongodb': True,
+        'with_postgresql': True,
+        'with_postgresql_extra': False,
+        'with_redis': True,
+        'with_grpc': True,
+        'with_clickhouse': True,
+        'with_rabbitmq': True,
+        'with_utest': True,
+        'namespace': 'userver',
+        'namespace_begin': 'namespace userver {',
+        'namespace_end': '}',
+        'mongo-c-driver/*:with_sasl': 'cyrus',
+    }
+
+    @property
+    def _source_subfolder(self):
+        return 'source'
+
+    @property
+    def _build_subfolder(self):
+        return os.path.join(self.build_folder, 'userver')
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires('boost/1.79.0', transitive_headers=True)
+        self.requires('c-ares/1.19.1')
+        self.requires('cctz/2.3', transitive_headers=True)
+        self.requires('concurrentqueue/1.0.3', transitive_headers=True)
+        self.requires('cryptopp/8.7.0')
+        self.requires('fmt/8.1.1', transitive_headers=True)
+        self.requires('libnghttp2/1.51.0')
+        self.requires('libcurl/7.88.1')
+        self.requires('libev/4.33')
+        self.requires('http_parser/2.9.4')
+        self.requires("openssl/[>=1.1 <4]")
+        self.requires('rapidjson/cci.20220822', transitive_headers=True)
+        self.requires('yaml-cpp/0.7.0')
+        self.requires("zlib/[>=1.2.11 <2]")
+
+        if self.options.with_jemalloc:
+            self.requires('jemalloc/5.3.0')
+        if self.options.with_grpc:
+            self.requires(
+                'grpc/1.54.3', transitive_headers=True, transitive_libs=True,
+            )
+            self.requires(
+                'googleapis/cci.20230501',
+                transitive_headers=True,
+                transitive_libs=True,
+            )
+            self.requires(
+                'grpc-proto/cci.20220627',
+                transitive_headers=True,
+                transitive_libs=True,
+            )
+            self.requires('protobuf/3.21.12') # override
```

Comment:
Can you give me a bit more insight into the comment here? Why is this requirement needed?

Reply:
It's not really needed